File: src/main/java/one/wabbit/parsing/Matcher.java
package one.wabbit.parsing;

import java.util.List;

public interface Matcher<S> {
    public int state();
    public void reset();
    public boolean nullable();
    public boolean failed();

    public int feed(S[] chunk, int from, int until);
    public int feed(List<S> chunk, int from, int until);
}

File: src/main/kotlin/one/wabbit/parsing/Matcher.kt
package one.wabbit.parsing

interface Matcher<S> {
    fun state(): Int
    fun reset()
    fun nullable(): Boolean
    fun failed(): Boolean

    fun feed(chunk: Array<S>, from: Int, until: Int): Int
    fun feed(chunk: List<S>, from: Int, until: Int): Int
}
